From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 19 Aug 2010 12:56:08 -0400
Subject: [misc] fix non-CONFIG_NUMA x86_64 compile
Message-id: <20100819125607.3195.82242.sendpatchset@prarit.bos.redhat.com>
Patchwork-id: 27718
O-Subject: [RHEL5 BZ 583673 PATCH]: Fix !CONFIG_NUMA compile
Bugzilla: 583673
RH-Acked-by: Don Zickus <dzickus@redhat.com>

We do not support custom-built kernels, however, the source should at least
compile.

cpu is defined without CONFIG_NUMA upstream so it seems like a simple thing to
do.

Successfully compiled by me.

Resolves BZ 583673.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86_64/kernel/setup.c b/arch/x86_64/kernel/setup.c
index 0a112ae..4ad790a 100644
--- a/arch/x86_64/kernel/setup.c
+++ b/arch/x86_64/kernel/setup.c
@@ -866,8 +866,8 @@ static void __init amd_detect_cmp(struct cpuinfo_x86 *c)
 {
 #ifdef CONFIG_SMP
 	unsigned bits;
-#ifdef CONFIG_NUMA
 	int cpu = smp_processor_id();
+#ifdef CONFIG_NUMA
 	int node = 0;
 	unsigned apicid = hard_smp_processor_id();
 #endif
